Fixed #36134 -- Add support for ABSENT ON NULL clause for JSONField. (#19097)
django/django: docs/ref/models/database-functions.txt

    ...         Lower("name"),
    ...         "alias",
    ...         F("age") * 2,
    ...         "goes_by",
    ...     )
    ... ).get()
    >>> author.json_array
    ['margaret smith', 'msmith', 50, null]

> If you want to omit null values, you can do so by passing ``absent_on_null=True``:

Reviewer: ```suggestion
Set the ``absent_on_null`` parameter to ``True`` to omit null values.
```
I would be tempted to combine the two usage examples so that the imports are not repeated etc. Like:

```
Accepts a list of field names or expressions and returns a JSON array
containing those values.

Set the ``absent_on_null`` parameter to ``True`` to omit null values.

Usage example:

.. code-block:: pycon

    >>> from django.db.models import F
    >>> from django.db.models.functions import JSONArray, Lower
    >>> Author.objects.create(name="Margaret Smith", alias="msmith", age=25)
    >>> author = Author.objects.annotate(
    ...     json_array=JSONArray(
    ...         Lower("name"),
    ...         "alias",
    ...         F("age") * 2,
    ...         "goes_by",
    ...     )
    ... ).get()
    >>> author.json_array
    ['margaret smith', 'msmith', 50, null]
    >>> author = Author.objects.annotate(
    ...     json_array=JSONArray(
    ...         Lower("name"),
    ...         "alias",
    ...         F("age") * 2,
    ...         "goes_by",
    ...         absent_on_null=True,
    ...     )
    ... ).get()
    >>> author.json_array
    ['margaret smith', 'msmith', 50]
```

.. versionadded:: 6.0

.. code-block:: pycon

    >>> from django.db.models import F
    >>> from django.db.models.functions import JSONArray, Lower
    >>> Author.objects.create(name="Margaret Smith", alias="msmith", age=25)
    >>> author = Author.objects.annotate(
    ...     json_array=JSONArray(
    ...         Lower("name"),
    ...         "alias",
    ...         F("age") * 2,
    ...         "goes_by",
    ...         absent_on_null=True,
    ...     )
    ... ).get()
    >>> author.json_array
    ['margaret smith', 'msmith', 50]

> .. admonition:: MySQL and MariaDB
> 
>     MySQL and MariaDB don't support ``True`` for the ``absent_on_null`` parameter.
> 
> .. admonition:: PostgreSQL
> 
>     ``True`` for the ``absent_on_null`` parameter requires PostgreSQL version 16
>     or later.

Reviewer: ```suggestion
.. admonition:: MySQL, MariaDB and PostgreSQL

    The ``absent_on_null`` parameter is ignored with MySQL, MariaDB and
    PostgreSQL < 16 as they do not support it.

.. versionchanged:: 6.0

    The ``absent_on_null`` parameter was added.
```
Then remove the `.. versionadded:: 6.0` note 
We also need a release note in the 6.0 notes

Author: Your documentation here doesn't actually reflect the behavior. It's not actually ignored, it raises a `NotSupportedError`. Is there a specific way I should phrase that?

Reviewer: ```
    MySQL, MariaDB and PostgreSQL < 16 do not support the ``absent_on_null``
    parameter.
```
perhaps :thinking:
